dotCMS/src/integration-test/java/com/dotcms/publishing/job/SiteSearchJobImplTest.java

```diff
@@ -448,74 +454,209 @@ public void Test_Incremental_Job_Test_Pages_Are_Found_Create_And_Publish_New_Pag
+            Language defaultLang = APILocator.getLanguageAPI().getDefaultLanguage();
+            Language secondLang = new LanguageDataGen().nextPersisted();
             folder = new FolderDataGen().site(site).nextPersisted();
 
-            Contentlet contentletLang1 = TestDataUtils
-                    .getEmployeeContent(true, lang1.getId(), null, site);
-
-            contentletLang1 = contentletAPI.find(contentletLang1.getInode(), systemUser, false);
-            contentletLang1.setStringProperty("firstName", "catherine");
-            contentletLang1 = contentletAPI.checkin(contentletLang1, systemUser, false);
-
-            ContentletDataGen.publish(contentletLang1);
+            Contentlet contentletDefaultLang = null;
+            Contentlet contentSecondLang = null;
 
-            Contentlet contentletLang2 = contentletAPI
-                    .find(contentletLang1.getInode(), systemUser, false);
-            contentletLang2.setInode("");
-            contentletLang2.setLanguageId(lang2.getId());
-            contentletLang2.setStringProperty("firstName", "catalina");
-            contentletLang2 = contentletAPI.checkin(contentletLang2, systemUser, false);
+            if(testCase.createContentInDefaultLanguage) {
+                contentletDefaultLang = createAndPublishEmployeeContent(site, defaultLang, "catherine");
 
-            ContentletDataGen.publish(contentletLang2);
-
-            final Container container = new ContainerDataGen().withContentType(contentletLang1
-                    .getContentType(), "$!{firstName}").nextPersisted();
-
-            ContainerDataGen.publish(container);
-
-            final String uuid = UUIDGenerator.generateUuid();
-
-            final Template template = new TemplateDataGen()
-                    .withContainer(container.getIdentifier(), uuid)
-                    .nextPersisted();
-
-            TemplateDataGen.publish(template);
-
-            HTMLPageAsset page = new HTMLPageDataGen(folder, template).languageId(lang1.getId())
-                    .nextPersisted();
-
-            HTMLPageDataGen.publish(page);
+                if(testCase.createContentInSecondLanguage) {
+                    createNewVersionAndPublishExistingEmployeeContent(secondLang, contentletDefaultLang,
+                            "catalina");
+                }
+            } else if(testCase.createContentInSecondLanguage) {
+                contentSecondLang = createAndPublishEmployeeContent(site, secondLang, "catalina");
+            }
 
-            final MultiTree multiTree = new MultiTree(page.getIdentifier(),
-                    container.getIdentifier(),
-                    contentletLang1.getIdentifier(), getDotParserContainerUUID(uuid), 0);
+            HTMLPageAsset pageDefaultLang = null;
+            HTMLPageAsset pageSecondLang = null;
 
-            APILocator.getMultiTreeAPI().saveMultiTree(multiTree);
+            Contentlet contentToPassToPage = contentletDefaultLang!=null
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'contentToPassToPage' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5303548)